app/services/work_packages/update_ancestors_service.rb

```diff
@@ -120,15 +120,43 @@ def derive_done_ratio(ancestor, loader)
   end
 
   def compute_derived_done_ratio(work_package, loader)
+    return if no_children?(work_package, loader)
+
+    case Setting.total_percent_complete_mode
+    when "work_weighted_average"
+      calculate_work_weighted_average_percent_complete(work_package)
+    when "simple_average"
+      calculate_simple_average_percent_complete(work_package, loader)
+    end
+  end
+
+  def calculate_work_weighted_average_percent_complete(work_package)
     return if work_package.derived_estimated_hours.nil? || work_package.derived_remaining_hours.nil?
     return if work_package.derived_estimated_hours.zero?
-    return if no_children?(work_package, loader)
 
     work_done = (work_package.derived_estimated_hours - work_package.derived_remaining_hours)
     progress = (work_done.to_f / work_package.derived_estimated_hours) * 100
     progress.round
   end
 
+  def calculate_simple_average_percent_complete(work_package, loader)
+    all_done_ratios = children_done_ratio_values(work_package, loader)
+
+    if work_package.done_ratio.present? && !work_package.status.excluded_from_totals
+      all_done_ratios << work_package.done_ratio
+    end
+
+    progress = all_done_ratios.sum.to_f / all_done_ratios.count
+    progress.round
```

Comment:
Found another issue: this fails if all work packages are excluded: the sum returns 0.0, count returns 0, so progress will be `NaN`, and `NaN.round` raises.

In this case, ∑ % complete should be cleared.

I'll fix that.

Reply:
Fixed in f0408482463